src/main/java/com/google/devtools/build/lib/analysis/starlark/FunctionTransitionUtil.java

```diff
@@ -407,36 +409,42 @@ private static void updateOutputDirectoryNameFragment(
     CoreOptions buildConfigOptions = toOptions.get(CoreOptions.class);
     Set<String> updatedAffectedByStarlarkTransition =
```

Comment:
Minor nit: would you mind documenting this line to make it clear what `updatedAffectedByStarlarkTransition` is meant for? It's deducible from the code but I think it helps to be super-explicit given the subtleties in this method.

Reply:
What would you think about me extracting a few of those lines into a separate method? That would mutate the global buildConfigOptions.affectedByStarlarkTransition and be invoked here as:

`updateAffectedByStarlarkTransition(buildConfigOptions, changedOptions);`

And document that method.

That should also minimize risk for confusion between buildConfigOptions.affectedByStarlarkTransition and updatedAffectedByStarlarkTransition.

Reply:
That's fair. Works for me.

Reply:
Resolved in 104f1dd